sql/api/src/main/scala/org/apache/spark/sql/errors/DataTypeErrors.scala

```diff
@@ -27,7 +27,7 @@ import org.apache.spark.unsafe.types.UTF8String
 /**
  * Object for grouping error messages from (most) exceptions thrown during query execution.
  * This does not include exceptions thrown during the eager execution of commands, which are
- * grouped into [[QueryCompilationErrors]].
```

Comment:
In the `sql/api` module, `QueryCompilationErrors` has been renamed to CompilationErrors`, so we will fix it here.

Reply:
We have both, AFAIK

Reply:
From the perspective of module dependency, it seems to express the meaning of `CompilationErrors` in the `sql/api` module, instead of `QueryCompilationErrors` in the `sql/catalyst` module.

Or we can write it as: `[[CompilationErrors]] and [[QueryCompilationErrors]]` ?

Reply:
The most primitive PR is here: https://github.com/apache/spark/pull/41794/files#diff-1b8affb347590e3ccabe359b07215c72aead8108ca3a5a020b3cb71e639c117bR25

Reply:
ok, let's leave the ref to `CompilationErrors` only.